arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -2547,4 +2580,71 @@ void RocksDBEngine::releaseTick(TRI_voc_tick_t tick) {
   }
 }
 
+bool RocksDBEngine::checkHealth() {
+  auto now = std::chrono::steady_clock::now();
+
+  // the following checks are executed under a mutex so that different
+  // threads can potentially call in here without messing up any data.
+  // in addition, serializing access to this function avoids stampedes
+  // with multiple threads trying to calculate the free disk space 
+  // capacity at the same time, which could be expensive.
+  MUTEX_LOCKER(guard, _healthCheckMutex);
+
+  bool lastCheckLongAgo = (now - _lastHealthCheckTimestamp > std::chrono::seconds(30));
```

Comment:
I doubt JPMC will like this. They create a ticket for their DB admin team for every ERROR log line we write. So if something happens, they wake up in the morning with lots of tickets. Maybe we have to build an exponential backoff here.

Reply:
At least the log message is always the same ID.

Reply:
Let's increase the 30 seconds to 24h.